### PR TITLE
[JSB] use key to schedule to avoid memory leak in scheduleOnce

### PR DESCRIPTION
**Pull request by pandamicro**
Re: cocos-creator/fireball#6418

Changes proposed in this pull request:
 * [JSB] use key to schedule to avoid memory leak in scheduleOnce

> Makes sure these boxes are checked before submitting your PR - thank you!
>
- [x] If your pull request has gone "stale", you should **rebase** your work on top of the latest version of the upstream branch.
- [x] If your commit history is full of small, unimportant commits (such as "fix pep8" or "update tests"), **squash** your commits down to a few, or one, discreet changesets before submitting a pull request.
- To official teams:
  - [x] Check that your javascript is following our [style guide](https://github.com/cocos-creator/fireball/blob/dev/.github/CONTRIBUTING.md) and end files with a newline
  - [x] Document new code with comments in source code based on [API Docs](https://github.com/cocos-creator/fireball#api-docs)
  - [x] Make sure any runtime log information in `cc.log` , `cc.error`, `cc.warn` or `cc.assert` has been moved into `DebugInfos.js` with an ID

@cocos-creator/engine-admins

### CODE REVIEW

File: jsb/jsb-etc.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ cc.js.mixin(cc.path, {
 });
 
 // cc.Scheduler
+var _callbackId = 0 | 998 * Math.random();
```
Comment (jareguo) [inline jsb/jsb-etc.js]: 为什么需要初始化为随机数呢？

Comment (dumganhar) [reply jsb/jsb-etc.js]: 同问，随机数不靠谱吧，如果随机到同一个值怎么办？

File: jsb/jsb-etc.js [hunk 2]
```diff
@@ -65,7 +66,13 @@ cc.Scheduler.prototype.schedule = function (callback, target, interval, repeat,
         repeat = isFinite(repeat) ? repeat : cc.macro.REPEAT_FOREVER;
     }
     delay = delay || 0;
-    this.scheduleCallbackForTarget(target, callback, interval, repeat, delay, paused);
+    var instanceId = target.__instanceId || target.uuid;
+    cc.assertID(instanceId, 1510);
+    if (!callback.__callbackId) {
+        callback.__callbackId = _callbackId++;
```
Comment (dumganhar) [inline jsb/jsb-etc.js:72]: 好吧，这里加加了，但是为什么初始化需要随机？

File: jsb/jsb-etc.js (hunk 1 above)
Comment (pandamicro) [reply jsb/jsb-etc.js]: 只是从 ClassManager 拿过来的代码，可以是 0

Comment (pandamicro) [reply jsb/jsb-etc.js]: 只是从 ClassManager 拿过来的代码，可以是 0

**Pull request merged**